File: 75SortColors/submit.java
public class Solution {
    // 三路快排
    public void sortColors(int[] nums) {
        int zero = -1; // [0,...zero] == 0
        int two = nums.length; // [two, ... nums.length) == 2

        for(int i = 0; i < two; ){
            if ( nums[i] == 0 ){ // 交换zero后面那个数与当前的nums[i]
                int temp = nums[++zero];
                nums[zero] = nums[i];
                nums[i++] = temp;
            }else if (nums[i] == 2) { // 交换two前面那个数与当前的nums[i]
                int temp = nums[--two];
                nums[two] = nums[i];
                nums[i] = temp;
            }else { // 为1的时候不动
                i++;
            }
        }
    }
}

File: 75SortColors/submit.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        # if len(nums) > 1:
        #     redCnt = 0
        #     whiteCnt = 0
        #     blueCnt = 0
        #     for n in nums:
        #         if n == 0:
        #             redCnt += 1
        #         if n == 1:
        #             whiteCnt += 1
        #         if n == 2:
        #             blueCnt += 1
        #     nums[:redCnt] = [0] * redCnt
        #     nums[redCnt: redCnt+whiteCnt] = [1] * whiteCnt
        #     nums[redCnt+whiteCnt:] = [2] * blueCnt
        if len(nums) > 1:
            cnt = [0] * 3
            
            for n in nums:
                assert n >= 0 and n <= 2
                cnt[n] += 1
                
            nums[:] = [0] * cnt[0] + [1] * cnt[1] + [2] * cnt[2]